src/seq_analysis/gc.rs

```diff
@@ -0,0 +1,33 @@
+// Copyright 2014-2016 Johannes Köster, Martin Larralde.
+// Licensed under the MIT license (http://opensource.org/licenses/MIT)
+// This file may not be copied, modified, or distributed
+// except according to those terms.
+
+// Sequence analysis algorithms such as orf finder or GC% counter
+
+// Maybe not the most clever way to do it, but at least runs in o(n)
+
+/// Base gc content counter
+fn gcn_content<'a>(sequence: &'a[u8], step: usize) -> f32 {
```

Comment:
I have added trait aliases that allow to easily use IntoIterator (see [here](http://rust-bio.github.io/rust-bio/bio/utils/index.html)). This way, you can get rid of the while loop. Using ``step`` from Itertools should be fine, and I don't think it will hurt performance.